### PR TITLE
Refund approval workflow not working for Transfer

### PR DESCRIPTION
**Pull request by allanmfx**
The isPushNeeded method does a small check for refund approval cases, but the receivePushCheckDuplicates method is not prepared to handle refund approvals for BankTransfer as it's ignoring it.

So I added a piece of code that will fix the validation for refund transactions.

### CODE REVIEW

**Comment by Buckaroo-Rens** [comment]
Thanks @allanmfx for your PR. I will let one of our developers check it a.s.a.p. Next time it would be nice if you can follow our guidelines regarding contributing, you can check them here: https://github.com/buckaroo-it/Magento2/blob/master/CONTRIBUTING.md 

Thanks again :)

**Comment by LucianTuriacArnia** [comment]
Your Pull Request it will skip all refunds from processing, for all payment methods. So I do not agree with the condition added by you. 

If you want to skip processing refund for Transfer add new condition in "isPushNeeded" method and ensure that you set a condition depending on the payment method, as we have for KlarnaKp.

MicrosoftTeams-image

Thank you,

**Comment by allanmfx** [comment]
@LucianTuriacArnia Don't think you correctly understood. I do not want to skip a refund transaction, the existing code itself is skipping refunds for Transfers with status 190. What I did is add an extra condition to the skip logic, if it's a refund do not skip the transaction if it's Transfer or Giftcard.

I did some tests on that and there are no side-effects or regressions.

**Comment by allanmfx** [comment]
@LucianTuriacArnia The current trace is:

1. Once a refund is approved on Buckaroo Plaza a 190 push is sent to Magento, then it will jump into `\Buckaroo\Magento2\Model\Push::receivePush`.
2. `isPushNeeded` will be called on https://github.com/buckaroo-it/Magento2/blob/master/Model/Push.php#L252
3. It will enter in the first IF block as it is a refund transaction (we can notice that the isPushNeeded indeed has a special treatment for refunds) and will check if the current transaction is 190 and if there's already a pending approval status, so `receivePushCheckDuplicates` is called at https://github.com/buckaroo-it/Magento2/blob/master/Model/Push.php#L252. Just to clarity why it checked for a pending_approval status and return false if this status has not been received: Because if there's no pending_approval it means the refund was successfully at the creditmemo creation time, so the push is not necessary as the return from Magento request to buckaroo already confirmed the refund.
4. Inside `receivePushCheckDuplicates` we can see that there are two payment methods in which validation is skipped: `$ignoredPaymentMethods`, so I'm ignoring the payment method check if it's a refund: `(!in_array($payment->getMethod(), $ignoredPaymentMethods) || $isRefund`

**Comment by LucianTuriacArnia** [comment]
@allanmfx I miss a bracket when I did code review, I will approve your PR.
image

**Review by LucianTuriacArnia — APPROVED** [review]

**Comment by LucianTuriacArnia** [comment]
Created new PR - https://github.com/buckaroo-it/Magento2/pull/920

**Pull request closed without merging**
